code/modules/cm_marines/radar.dm

```diff
@@ -235,3 +235,34 @@
 			if(uniform.has_sensor && uniform.sensor_mode >= SENSOR_MODE_LOCATION) // Suit sensors must be on maximum
 				return TRUE
 	return FALSE
+
+
+//Synthetic K9 Scent Tracking, allows K9s to track CLF, UPP, etc as well as Preds... optic camo can't hide the fact you stink!
+/datum/radar/scenttracker/find_atom()
+	return locate(selected) in GLOB.human_mob_list
+
+/datum/radar/scenttracker/scan()
+	. = ..()
+	objects = list()
+	for(var/i in GLOB.human_mob_list)
+		var/mob/living/carbon/human/humanoid = i
```

Comment:
Condense this into one line, as anything

Reply:
single line in the for() checks out as expected